apps/managedidentity/managedidentity.go

```diff
@@ -508,8 +508,17 @@ func (c Client) retry(maxRetries int, req *http.Request) (*http.Response, error)
 		if err == nil && !contains(retrylist, resp.StatusCode) {
 			return resp, nil
 		}
+		// For IMDS, use exponential backoff based on attempt number
+		var waitTime time.Duration
+		if c.source == DefaultToIMDS {
+			// Exponential backoff with base of 1 second: 1s, 2s, 4s, 8s, etc.
```

Comment:
This is less backoff than recommended: https://learn.microsoft.com/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance

Reply:
Updated the time to reflect the time.